### PR TITLE
DL: Add alternative options for compile params

### PR DESCRIPTION
**Pull request by orhankislal**
Add support for optimizer (string) and loss (objective function)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/822/

File: src/ports/postgres/modules/deep_learning/madlib_keras_wrapper.py_in [hunk 1]
```diff
@@ -120,15 +122,27 @@ def parse_compile_params(str_of_args):
...
+    else:
+        opt_params = compile_dict['optimizer'].split('(')[1][:-1]
+        opt_params_array = opt_params.split(',')
+        opt_params_clean = map(split_and_strip, opt_params_array)
+        key_value_params = { x[0] : x[1] for x in opt_params_clean}
+        final_args = { key: bool(value) if value == 'True' or value == 'False'
+                       else float(value)
+                       for key,value in key_value_params.iteritems() }
+
+    loss_split = compile_dict['loss'].split('.')
```
[the remark this answers is not in the document]
Comment (orhankislal) [reply src/ports/postgres/modules/deep_learning/madlib_keras_wrapper.py_in]: If we want to support individual loss functions, we have to import every single one. It is possible that future versions of keras might change these or add new ones. Instead of trying to update the code with every keras change, we can import the losses as a whole and direct the user to use it.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/831/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/834/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/835/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/836/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/837/
